scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -320,7 +320,10 @@ class FossId(
                 checkNotNull(scans)
 
                 val existingScan = scans.sortedByDescending { it.id }.find { scan ->
-                    scan.gitBranch == revision && scan.gitRepoUrl == url
+                    // The scans in the server contain the url with the credentials so we have to remove it for the
+                    // comparison. If we don't, the scans won't be matched if the password changes!
+                    val urlWithoutCredentials = scan.gitRepoUrl!!.substringAfter("@")
```

Comment:
This also strips the protocol part, so does the comparison really work? Can't / shouldn't we make use of `replaceCredentialsInUri()` instead?